http-client-core/src/main/java/io/micronaut/http/client/StreamingHttpClient.java

```diff
@@ -42,6 +42,18 @@
      */
     <I> Publisher<ByteBuffer<?>> dataStream(HttpRequest<I> request);
 
+
+    /**
+     * Request a stream of data where each emitted item is a {@link ByteBuffer} instance.
+     *
+     * @since 3.1.0
+     * @param request   The request
+     * @param errorType The type that the response body should be coerced into if the server responds with an error
+     * @param <I>       The request body type
+     * @return A {@link Publisher} that emits a stream of {@link ByteBuffer} instances
+     */
+    <I> Publisher<ByteBuffer<?>> dataStream(HttpRequest<I> request, Argument<?> errorType);
```

Comment:
I know it is missing from the rest of the methods, but it seems this whole API is missing `@Nullable` / `@NonNull` the presence of which would help kotlin users